insights/utils.py

```diff
@@ -74,3 +76,9 @@ class InsightsSettings:
     @classmethod
     def get(cls, key):
         return frappe.db.get_single_value("Insights Settings", key)
+
+
+def detect_encoding(file_path: pathlib.Path):
```

Comment:
Taking a PathLike argument is better here, user should not have to import `Path` from pathlib to use this util. Maybe you can then use Path here to create an instance and call .open on it.